### PR TITLE
Render custom message when user coming from legacy after dataset creation

### PR DESCRIPTION
**Pull request by laurentqro**
When a user creates a new dataset on legacy, they are redirected to the corresponding dataset page on Find.

However, because legacy -> Find sync is not instantaneous and we cannot modify legacy, we render a page informing the user their dataset will be available for view soon.

https://trello.com/c/A1LTiZQS/316-display-a-message-to-publishers-after-dataset-creation-m

### CODE REVIEW

File: app/controllers/legacy/datasets_controller.rb [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+  def available_soon
+    if request_originates_from_legacy?
+      render 'datasets/available_soon'
+    else
+      raise
+    end
+  end
+
+  def request_originates_from_legacy?
+    request.referrer.present? && URI.parse(request.referrer).host == ENV['LEGACY_HOST']
```
Comment (andrewgarner) [inline app/controllers/legacy/datasets_controller.rb]: Find and legacy will share the same host. I think we'll need to look at the path of the referrer.

File: app/views/datasets/_additional_info.html.erb [hunk 2]
```diff
@@ -61,23 +61,6 @@
               <dd><%= i['metadata_language'] %></dd>
             <% end %>
           </dl>
-
-          <!--
```
Comment (andrewgarner) [inline app/views/datasets/_additional_info.html.erb]: How is this related to this PR?

Comment (laurentqro) [reply app/views/datasets/_additional_info.html.erb]: it isn't, just removing dead comments
